Apps/W1/HybridGP/app/src/Migration/History/GPPopulateHistTables.Codeunit.al

```diff
@@ -1106,13 +1145,46 @@ codeunit 40900 "GP Populate Hist. Tables"
             HistInventoryTrxLine."Transfer To Location Code" := GPIVTrxAmountsHist.TRNSTLOC;
             HistInventoryTrxLine."Reason Code" := GPIVTrxAmountsHist.Reason_Code;
 
-            if not HistInventoryTrxLine.Insert() then
+            if HistInventoryTrxLine.Insert() then
+                PopulateInventoryTrxLineSerialLotDetail(HistInventoryTrxLine, GPIVTrxAmountsHist)
+            else
                 ReportLastError(Database::GPIVTrxAmountsHist, GPIVTrxAmountsHist.DEX_ROW_ID, "Hist. Migration Step Type"::"GP Inventory Trx.", DocumentNo);
 
             AfterProcessedNextChildRecord();
         until GPIVTrxAmountsHist.Next() = 0;
     end;
 
+    local procedure PopulateInventoryTrxLineSerialLotDetail(var HistInventoryTrxLine: Record "Hist. Inventory Trx. Line"; var GPIVTrxAmountsHist: Record GPIVTrxAmountsHist)
+    var
+        GPIVSerialLotNumberHist: Record GPIVSerialLotNumberHist;
+        HistInvtTrxSerialLot: Record "Hist. Invt. Trx. SerialLot";
+    begin
+        GPIVSerialLotNumberHist.SetLoadFields(IVDOCTYP, DOCNUMBR, LNSEQNBR, SERLTNUM, SERLTQTY, FROMBIN, TOBIN, MFGDATE, EXPNDATE, TRXSORCE);
```

Comment:
In general this is not needed for these tables. It can also mess up the extensibility story. See if you really need to limit the fields that are set.